src/tiledb/cloud/asset.py

```diff
@@ -38,3 +118,109 @@ def info(uri: str) -> Union[ArrayInfo, GroupInfo]:
     asset_type: str = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
     func = info_map[asset_type]
     return func(uri)
+
+
+def update_info(
+    uri: str,
+    *,
+    description: Optional[str] = None,
+    name: Optional[str] = None,
+    tags: Optional[list[str]] = None,
+    access_credentials_name: Optional[str] = None,
+) -> None:
+    """
+    Update asset info settings.
+
+    :param uri: tiledb URI of the asset.
+    :type uri: str
+    :param description: Asset description, defaults to None
+    :type description: str
+    :param name: Asset name, defaults to None
+    :type name: str
+    :param tags: Asset tags, defaults to None
+    :type tags: list[str]
+
+    For example, to update only the description of an asset, call update_info()
+    with only a 'description' keyword argument:
+
+    >>> from tiledb.cloud import asset
+    >>> asset.update_info(
+    ...     "tiledb://your-namespace/your-asset",
+    ...     description="This is a new description"
+    ... )
+    """
+    # Note: logo not yet supported.
+    asset_type: str = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
+    if asset_type == "array":
+        return array.update_info(
+            uri,
+            array_name=name,
+            description=description,
+            access_credentials_name=access_credentials_name,
+            tags=tags,
+        )
+    elif asset_type == "group":
+        return groups.update_info(uri, description=description, name=name, tags=tags)
+    else:
+        raise tiledb.TileDBError(f"Invalid asset type '{asset_type}'")
+
+
+def list_shared_with(uri: str) -> None:
+    """List an asset's sharing policies.
+
+    :param uri: tiledb URI of the asset.
+    :type uri: str
+    :return: a list of ArraySharing or GroupSharing objects.
+    """
+    sharing_map: Mapping[str, Callable] = {
+        "array": array.list_shared_with,
+        "group": groups.list_shared_with,
+    }
+    asset_type: str = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
+    func = sharing_map[asset_type]
+    return func(uri)
+
+
+def share(
+    uri: str, namespace: str, permissions: Optional[Union[str, list[str]]] = "read"
```

Comment:
I could be easily talked out of providing a default for permissions.

Reply:
I think default here is good. Additionally at some point we need to coordinate programmatic vs ui. UI uses "roles" which they translate internally to these permissions.